sources/Scripts/selfReferencingLinks.py

```diff
@@ -14,21 +14,31 @@ def linkSelfReferences():
     listOfTosafotRefWithSource = findEverySelfReference(tosRefs)
     listOfRashiLinks = createLinks(listOfRashiRefsWithSource)
     listOfTosafotLinks = createLinks(listOfTosafotRefWithSource)
-    functions.post_link(listOfRashiLinks)
-    functions.post_link(listOfTosafotLinks)
 
-    print(listOfRashiLinks)
-    print(listOfTosafotLinks)
-    # print(len(listOfRashiLinks))
-    # print(len(listOfTosafotLinks))
+
+    brokenLinks = open('C:\Users\joshuaegoldmeier\Documents\BrokenLinks.txt','a')
+
+    for eachLink in listOfRashiLinks:
+        try:
+            functions.post_link(eachLink)
+        except Exception:
+            brokenLinks.write(eachLink)
+    for eachLink in listOfTosafotLinks:
+        try:
+            functions.post_link(eachLink)
```

Comment:
Try to upload an array of links, rather than one at a time as HTTP requests are very slow. Best would be not to add bad refs to the array in the first place, otherwise, you can use the del command to remove a value from the middle of the list, or pop to remove from the end. Keep in mind - the del command is O(n), whereas append and pop are approximately O(1) - so keep that in mind when deciding where and how to remove bad refs.